tests/testthat/test-query-search.R

```diff
@@ -993,3 +995,31 @@ test_that("query with mixed types returns results with valid comparison", {
     orderly_search(quote(parameter:a > 1), root = root),
     x)
 })
+
+
+test_that("&& takes precedence over ||", {
```

Comment:
I am not sure that this is precendence but associativity - https://en.wikipedia.org/wiki/Operator_associativity Python does look to have the same associativity here which is good.

One thing that we could do is to spot things like A && B || C and require disambiguation (after the parse phase). So allow any number of the _same_ operator chained (`A || B || ... || Z` or `A && B && ... && Z`) but within a chain require that any switch between the two is grouped with parentheses. Some C linters will require this

Reply:
I think it is precedence no? From that article looks like if && and || were right-to-left associative we would have
`A && B || C` be interpreted as `A && (B || C)` but this is saying `&&` always takes precedence so it is `(A && B) || C` and in the case `A || B && C` it is `A || (B && C)`

Like this table for C https://en.wikipedia.org/wiki/Operators_in_C_and_C%2B%2B#Operator_precedence

If we didn't allow it then it would mean it was not associative?

Reply:
You're right! See `?Syntax` for the full ordering. Looks like python does the same: https://www.programiz.com/python-programming/precedence-associativity